### PR TITLE
earlyoom: patch absolute dbus path and make nixos module up to date

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change
`earlyoom` v1.6 replaces old notification logic with `dbus-send` / `systembus-notify`. (See [Changelog](https://github.com/rfjakob/earlyoom/tree/v1.6#changelog).)
But dbus path is not patched and options of nixos/earlyoom is not up to date.

Related: #88441

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - Not changed.
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/system/earlyoom.nix [hunk 1]
```diff
@@ -67,9 +67,19 @@ in
...
+        '';
+      };
+
+      enableNotifications = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Send notifications about killed processes via the system d-bus.
+          To actually see the notifications in your GUI session, you need to have
+          <literal>systembus-notify</literal> running as your user.
```
Comment (matthewbauer) [inline nixos/modules/services/system/earlyoom.nix:82]: Maybe we should package this? Opened https://github.com/rfjakob/systembus-notify/issues/3 for a user service

Comment (oxalica) [reply nixos/modules/services/system/earlyoom.nix:82]: See #88441. But is there any way to setup a systemd user service in system configuration?

Comment (matthewbauer) [reply nixos/modules/services/system/earlyoom.nix:82]: It sounds like the autostart desktop file is enough

Comment (matthewbauer) [reply nixos/modules/services/system/earlyoom.nix:82]: (but yes there is, through systemd.user.services I think)

Comment (oxalica) [reply nixos/modules/services/system/earlyoom.nix:82]: Should I make `environment.systemPackages = [ pkgs.systembus-notify ];` when notification is enabled? But this requires #88441 to be merged first.

Comment (matthewbauer) [reply nixos/modules/services/system/earlyoom.nix:82]: Yeah - let’s do that and merge systembus-notify first

**Pull request merged**